### PR TITLE
Add CheckState/Validate for PackageResource

### PR DESCRIPTION
**Pull request by adjackura**

### CODE REVIEW

File: config/package_resouce.go [hunk 1]
```diff
@@ -82,164 +92,264 @@ func (p *packageResouce) validate() (*ManagedResources, error) {
...
+		cache = &rpmInstalled
+		refreshFunc = packages.InstalledRPMPackages
+	default:
+		return fmt.Errorf("unknown or unpopulated ManagedPackage package type: %+v", mp)
+	}
+
+	// Cache already populated.
+	if *cache != nil {
+		return nil
+	}
```
Comment (manjunathganga) [inline config/package_resouce.go:201]: When the cache is already populated, when do we re-populate it?

If a package is installed outside the config management system, would the cache reflect that?

Comment (adjackura) [reply config/package_resouce.go:201]: I was going back and forth on how to implement this, for now i created a catch type with a refreshed timestamp

File: config/package_resource.go [hunk 2]
```diff
@@ -195,8 +201,8 @@ func populateInstalledCache(ctx context.Context, mp ManagedPackage) error {
 		return fmt.Errorf("unknown or unpopulated ManagedPackage package type: %+v", mp)
 	}
 
-	// Cache already populated.
-	if *cache != nil {
+	// Cache already populated within the last 3 min.
+	if cache.cache != nil && cache.refreshed.After(time.Now().Add(-3*time.Minute)) {
```
Comment (manjunathganga) [inline config/package_resource.go:205]: Where is this field getting updated?

Comment (adjackura) [reply config/package_resource.go:205]: it wasnt, just fixed

**Review by manjunathganga — APPROVED** [review]

**Pull request merged**